File: windows/Models/Goal.cs
namespace Tick.Models;

/// <summary>
/// 目标：顶层组织单元。
/// 所有字段使用可空或带默认值，保证数据库迁移与旧数据兼容。
/// </summary>
public class Goal
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public string Name { get; set; } = "";

    /// <summary>颜色 HEX 字符串（如 "#000000"；"auto" = 自动：深色模式白 / 浅色模式黑）</summary>
    public string ColorHex { get; set; } = "auto";

    /// <summary>图标名（Segoe Fluent Icons 字形；null = 未设置）</summary>
    public string? IconSystemName { get; set; }

    /// <summary>开始日期（null = 未设置）</summary>
    public DateTime? StartDate { get; set; }

    /// <summary>截止日期（null = 未设置）</summary>
    public DateTime? EndDate { get; set; }

    /// <summary>开始时间是否精确到小时</summary>
    public bool StartDatePreciseToHour { get; set; }

    /// <summary>截止时间是否精确到小时（影响倒计时是否显示「时」）</summary>
    public bool EndDatePreciseToHour { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.Now;

    /// <summary>进度统计模式原始字符串（枚举存 raw string）</summary>
    public string ProgressCountingModeRaw { get; set; } = ProgressCountingMode.AllTasks.ToString();

    /// <summary>进度统计模式（读写映射枚举，未知／未设置回退 allTasks）</summary>
    public ProgressCountingMode ProgressCountingMode
    {
        get => Enum.TryParse<ProgressCountingMode>(ProgressCountingModeRaw, out var mode) ? mode : ProgressCountingMode.AllTasks;
        set => ProgressCountingModeRaw = value.ToString();
    }

    /// <summary>目标下的一级任务（删除目标时级联删除全部任务）</summary>
    public List<TaskItem> Tasks { get; set; } = new();

    public Goal() { }

    public Goal(string name,
                string colorHex = "auto",
                string? iconSystemName = null,
                DateTime? startDate = null,
                DateTime? endDate = null,
                bool startDatePreciseToHour = false,
                bool endDatePreciseToHour = false,
                ProgressCountingMode progressCountingMode = ProgressCountingMode.AllTasks)
    {
        Name = name;
        ColorHex = colorHex;
        IconSystemName = iconSystemName;
        StartDate = startDate;
        EndDate = endDate;
        StartDatePreciseToHour = startDatePreciseToHour;
        EndDatePreciseToHour = endDatePreciseToHour;
        ProgressCountingMode = progressCountingMode;
    }
}

File: android/app/src/main/java/com/tick/app/android/model/Goal.kt
package com.tick.app.android.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.UUID

/**
 * 目标：顶层组织单元。进度统计模式以 raw string 存储，未知/未设置回退 allTasks。
 */
@Entity(tableName = "goals")
data class Goal(
    @PrimaryKey val id: String = UUID.randomUUID().toString(),
    val name: String,
    /** 颜色 HEX 字符串（如 "#000000"；"auto" = 深色白 / 浅色黑） */
    val colorHex: String = "auto",
    /** 图标名（null = 未设置） */
    val iconSystemName: String? = null,
    /** 开始日期 epoch millis（null = 未设置） */
    val startDate: Long? = null,
    /** 截止日期 epoch millis（null = 未设置） */
    val endDate: Long? = null,
    /** 开始时间是否精确到小时 */
    val startDatePreciseToHour: Boolean = false,
    /** 截止时间是否精确到小时 */
    val endDatePreciseToHour: Boolean = false,
    val createdAt: Long = System.currentTimeMillis(),
    /** 进度统计模式原始值（未设置回退 allTasks） */
    val progressCountingModeRaw: String = ProgressCountingMode.ALL_TASKS.raw
)

/** 进度统计模式（读写映射，未知/未设置回退 allTasks） */
val Goal.progressCountingMode: ProgressCountingMode
    get() = ProgressCountingMode.fromRaw(progressCountingModeRaw)